### PR TITLE
store, o/snapstate: send default-tracks header, use RedirectChannel

### PR DESCRIPTION
**Pull request by chipaca**
This implements default tracks on install.

### CODE REVIEW

File: overlord/snapstate/snapstate_test.go [hunk 1]
```diff
@@ -2057,6 +2057,37 @@ func (s *snapmgrTestSuite) TestInstallStrictIgnoresClassic(c *C) {
 	c.Check(snapst.Classic, Equals, false)
 }
 
+func (s *snapmgrTestSuite) TestInstallPotato(c *C) {
```
Comment (pedronis) [inline overlord/snapstate/snapstate_test.go]: s/Potato/SnapWithDefaultTrack/ ?

File: overlord/snapstate/snapstate.go [hunk 2]
```diff
@@ -823,8 +828,13 @@ func InstallMany(st *state.State, names []string, userID int) ([]string, []*stat
 			return nil, nil, err
 		}
 
+		channel := "stable"
+		if sar.RedirectChannel != "" {
+			channel = sar.RedirectChannel
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go]: is this bit unit tested?

File: store/store.go [hunk 3]
```diff
@@ -929,6 +929,7 @@ func (s *Store) newRequest(ctx context.Context, reqOptions *requestOptions, user
 	req.Header.Set(hdrSnapDeviceArchitecture[reqOptions.APILevel], s.architecture)
 	req.Header.Set(hdrSnapDeviceSeries[reqOptions.APILevel], s.series)
 	req.Header.Set(hdrSnapClassic[reqOptions.APILevel], strconv.FormatBool(release.OnClassic))
+	req.Header.Set("Snap-Device-Capabilities", "default-tracks")
```
Comment (pedronis) [inline store/store.go]: this bit doesn't seem to be unit tested

File: overlord/snapstate/snapstate_test.go (hunk 1 above)
Comment (chipaca) [reply overlord/snapstate/snapstate_test.go]: whoops

**Comment by chipaca** [comment]
I'll push additional unit tests this PM

File: overlord/snapstate/snapstate.go (hunk 2 above)
Comment (chipaca) [reply overlord/snapstate/snapstate.go]: I added an InstallMany unit test that checks this.

File: store/store.go (hunk 3 above)
Comment (chipaca) [reply store/store.go]: now it is :-) thank you

File: overlord/snapstate/snapstate_test.go (hunk 1 above)
Comment (chipaca) [reply overlord/snapstate/snapstate_test.go]: fixed

**Pull request merged**